precompiles/outposts/stride/types.go

```diff
@@ -94,18 +95,68 @@ func parseLiquidStakeArgs(args []interface{}) (common.Address, common.Address, *
 	return sender, token, amount, receiver, nil
 }
 
-// CreateMemo creates the memo for the StakeIBC actions - LiquidStake and Redeem.
-func CreateMemo(action, receiverAddress string) (string, error) {
+// parseRedeemStakeArgs parses the arguments from the Redeem Stake method call
+func parseRedeemStakeArgs(args []interface{}) (common.Address, common.Address, common.Address, string, *big.Int, error) {
+	if len(args) != 5 {
+		return common.Address{}, common.Address{}, common.Address{}, "", nil, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 5, len(args))
+	}
+
+	sender, ok := args[0].(common.Address)
+	if !ok {
+		return common.Address{}, common.Address{}, common.Address{}, "", nil, fmt.Errorf(cmn.ErrInvalidType, "sender", common.Address{}, args[0])
+	}
+
+	receiver, ok := args[1].(common.Address)
+	if !ok {
+		return common.Address{}, common.Address{}, common.Address{}, "", nil, fmt.Errorf(cmn.ErrInvalidType, "receiver", common.Address{}, args[1])
+	}
+
+	token, ok := args[2].(common.Address)
+	if !ok {
+		return common.Address{}, common.Address{}, common.Address{}, "", nil, fmt.Errorf(cmn.ErrInvalidType, "token", common.Address{}, args[2])
+	}
+
+	amount, ok := args[3].(*big.Int)
+	if !ok {
+		return common.Address{}, common.Address{}, common.Address{}, "", nil, fmt.Errorf(cmn.ErrInvalidType, "amount", big.Int{}, args[3])
+	}
+
+	strideForwarder, ok := args[4].(string)
+	if !ok {
+		return common.Address{}, common.Address{}, common.Address{}, "", nil, fmt.Errorf(cmn.ErrInvalidType, "strideForwardeer", "", args[4])
```

Comment:
should use string as type. You could also use `%T` to get the original type of the `args[4]` argument